### PR TITLE
Support dynamic fontsizes in iOS devices

### PR DESCRIPTION
**Pull request by ssetem**
### WHY are these changes introduced?

Currently Webviews & Mobile Web in iOS don't respect the users preference for Font Scale which is quite inaccessible.

This PR introduces the use of font: -apple-system-body if it is supported.

Inspired by this [dev.to article](https://dev.to/colingourlay/how-to-support-apple-s-dynamic-text-in-your-web-content-with-css-40c0#:~:text=If%20you've%20never%20seen,apple%2Dsystem%2Dbody%20)

### WHAT is this pull request doing?

<details>
  <summary>Before</summary>
  
</details>
<details>
  <summary>After</summary>

</details>

### How to 🎩

1. Open iOS simulator
2. Open Safari within the iOS Simulator
3. Open polaris storybook using this url https://5d559397bae39100201eedc1-krdvxqhrcy.chromatic.com/?path=/story/all-components-text--variants
4. Change Fontscale in `Settings > Accessibility > Display & Text Size > Larger Text ` (using the slider at the bottom)
5. Switch back to Safari
6. Verify the font scale has updated to match the preference in accessibility

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by fortmarek — APPROVED** [review]
Tested this on my simulator and can confirm this works: https://screenshot.click/23-59-152vf-zw6aq.mp4 ❤️

File: .changeset/olive-fishes-provide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+supports dynamic fontsizes for iOS
```
Comment (sam-b-rose) [inline .changeset/olive-fishes-provide.md]:
```suggestion
Added support for [Dynamic Type](https://developer.apple.com/documentation/uikit/uifont/scaling_fonts_automatically) in iOS
```

**Review by sam-b-rose — APPROVED** [review]
Looks good. Also verified in iOS simulator 👍

**Review by kyledurand — APPROVED** [review]
Looks good 👍 Solid a11y improvement, thanks!

**Pull request merged**